console/console-init/ui/src/modules/iot-device/components/AddGateways/AddGateway.tsx

```diff
@@ -52,54 +98,142 @@ export const AddGateways: React.FunctionComponent<IGatewaysProps> = ({
+    }
+  };
+
+  const onChangeInput = async (value: string) => {
+    // TODO: Use the backend query to populate opions in TypeAhead.
+    return [
+      {
+        id: "id-juno:57966",
+        isDisabled: false,
+        key: "key-juno:57966",
+        value: "juno:57966"
+      }
+    ];
+  };
+
+  const clearSelection = () => {
+    setSelectedGateways([]);
+  };
 
   const onChangeID = (id: string) => {
-    deviceIDRegExp.test(id) ? setIsValid(true) : setIsValid(false);
+    // TODO: Use the correct regular expression for Device ID.
+    // deviceIDRegExp.test(id) ? setIsValid(true) : setIsValid(false);
+    if (id?.length > 1) {
+      setIsValid(true);
+    }
     setInputID(id);
   };
 
   return (
     <Grid>
       <GridItem span={8}>
-        <Title id="ag-device-info-title" headingLevel="h3" size="2xl">
+        <Title id="add-gateway-device-info-title" headingLevel="h3" size="2xl">
           {header}
         </Title>
         <br />
-        <DeviceListAlert
-          visible={true}
-          isInline={true}
-          variant={"info"}
-          title="The concept of AMQ IoT gateway"
-          description={alertDescription}
-        />
+
+        <Popover
+          bodyContent={
+            <div>
+              In AMQ IoT, gateway devices are represented in Hono in the same
+              way as any other devices.
+            </div>
+          }
+          aria-label="Add gateway devices info popover"
+          closeBtnAriaLabel="Close Gateway Devices info popover"
+        >
+          <Button
+            variant="link"
+            id="add-gateway-info-popover"
+            icon={<InfoCircleIcon />}
+          >
+            How AMQ IoT handles gateways?
+          </Button>
+        </Popover>
+        <br />
         <br />
         <Form>
           <FormGroup
             label="Device ID of gateway"
-            type="number"
-            id="ag-form-group"
+            id="add-gateway-form-group"
             isRequired
-            helperTextInvalid="Age has to be a number"
+            helperTextInvalid="Invalid device ID"
             fieldId="add-gateway-device-id-input"
             validated="default"
           >
             <Flex>
               <FlexItem>
-                <TextInput
-                  validated="default"
-                  value={inputID}
-                  id="add-gateway-device-id-input"
-                  aria-label="Input device ID"
-                  aria-describedby="Input device ID"
-                  onChange={onChangeID}
+                <DropdownWithToggle
+                  id="add-gateway-type-dropdown"
+                  toggleId="add-gateway-type-dropdown-toggle"
+                  position={DropdownPosition.left}
+                  dropdownItems={gatewayTypeOptions}
+                  onSelectItem={onTypeSelect}
+                  value={inputType}
                 />
               </FlexItem>
+              <FlexItem>
+                {inputType === "Device ID" ? (
```

Comment:
wrong check syntax. inputType value should not be like label. It should be in key/aatribute format.
device_id or deviceId etc.
Please revisit this.